### PR TITLE
Truncate Dialog File Name and Extension

### PR DESCRIPTION
**Pull request by stuartmscott**
### Description:
Fixes file dialog text issues with truncation

Fixes https://gophers.slack.com/archives/CK5U4BU87/p1588630548028400

Screen Shot 2020-05-04 at 6 24 09 PM

Screen Shot 2020-05-04 at 6 21 00 PM

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

#### Where applicable:

- [ ] Public APIs match existing style.
- [ ] Any breaking changes have a deprecation path or have been discussed.

### CODE REVIEW

**Comment by stuartmscott** [comment]
Extension still extends beyond the icon file bounds resulting in grey-on-blue text. 10 is also quite an arbitrary number and may not work at all resolutions. Unfortunately canvas.Text does not implement text wrapping. widget.Label has text wrapping, but it renders text in theme.TextColor and does not support custom text color.

File: dialog/file.go [hunk 1]
```diff
@@ -42,9 +42,12 @@ func (f *fileDialog) makeUI() fyne.CanvasObject {
 				f.open.Enable()
 			}
 		}
+		saveName.Wrapping = fyne.TextTruncate
```
Comment (andydotxyz) [inline dialog/file.go:45]: We should not truncate an entry, the user needs to be able to edit the whole thing. A ScrollContainer is probably the right solution.

File: dialog/fileitem.go [hunk 2]
```diff
@@ -57,6 +57,9 @@ func fileParts(path string) (name, ext string) {
 	if len(ext) > 1 {
 		ext = ext[1:]
 	}
+	if len(ext) > 10 {
```
Comment (andydotxyz) [inline dialog/fileitem.go:60]: From your screenshots it looks unlikely that more than 5 characters will fit... it’s probably better to have it sometimes truncates too much (iiiii) than still overlap at times (mmmmmmmmmm). I’d also consider defining a local const or naming a variable for this number.

File: dialog/file.go (hunk 1 above)
Comment (stuartmscott) [reply dialog/file.go:45]: Ack

File: dialog/fileitem.go (hunk 2 above)
Comment (stuartmscott) [reply dialog/fileitem.go:60]: Yeah okay. 5 is another arbitrary number and may not work for all resolutions but you're right that it is better to over truncate than spill over into other elements. I can't think of a better solution

**Comment by stuartmscott** [comment]
5 char limit looks better than 10

Screen Shot 2020-05-05 at 8 12 47 AM

Unfortunately there is something weird happening with the scroller - it doesn't initially respond to events, but then suddenly works. @andydotxyz can you try locally?

**Comment by stuartmscott** [comment]
Also the scroll bar is far too thick

File: dialog/fileitem.go (hunk 2 above)
Comment (andydotxyz) [reply dialog/fileitem.go]: I agree it is arbitrary, but not sure what you mean about not working on some resolutions? This is all scaled proportionately so the only thing that would change the proportions is a custom theme that changes font size or icon size.

**Comment by andydotxyz** [comment]
I can't seem to replicate the issue with scrolling - perhaps you can provide more detailed steps as there are now a few scrollers on the screen. Additionally I expect there could be a difference between initial state and loading other directories / file names.

The entry scroll bar works as expected here - that large size is the hover state. Was that showing by default at your end?

File: dialog/fileitem.go (hunk 2 above)
Comment (stuartmscott) [reply dialog/fileitem.go]: Yup, I was thinking that really small screens, or really large screens could surface the limitations of this approach, but you're right that it _should_ all get scaled properly. Custom themes pose more of an issue, but I don't think we can protect against all customizations. This arbitrary number fixes _most_ issues, so let's go with it for now, and we can always improve it later

**Comment by stuartmscott** [comment]
I captured a video, the first part of it I am interacting with the scroller and scrollbar but does not respond. Then it suddenly starts to work :/

I should turn on visible taps and scrolls so you can see, but rest assured I'm tapping, scrolling, and dragging!

Open Dialog
out

Save Dialog
This seems to not work, until I tap the entry and it gains focus, then scroll works
out

**Comment by andydotxyz** [comment]
Sorry I realised this one got left behind. Is there something additional we need to do with the scrolling issue, or can it be replicated elsewhere?

**Comment by stuartmscott** [comment]
I'm still trying to figure out the scroll issue, but if it is related to the scroll issue in #930 then I think we can get this in and fix the issue separately.

**Comment by andydotxyz** [comment]
Hmm, I wondered the same but it might be wishful thinking as the content looks correct in this case.

**Comment by andydotxyz** [comment]
I’d recommend rebasing on develop just in case :)

**Comment by stuartmscott** [comment]
You're right, I could not repro after merging develop

**Comment by andydotxyz** [comment]
Good to merge then?

**Comment by andydotxyz** [comment]
Green from Travis

**Pull request merged**